### ISSUE TITLE
[valid-mock-module-path] false positive when using virtual mocks

### ISSUE THREAD

**Issue by Xunnamius** [jest-community/eslint-plugin-jest#1945]
Jest supports [virtual mocks](https://jestjs.io/docs/jest-object#jestmockmodulename-factory-options), or mocking modules/paths that do not exist anywhere in the system.

```ts
jest.mock(
  '../moduleName',
  () => {
    /*
     * Custom implementation of a module that doesn't exist in JS,
     * like a generated module or a native module in react-native.
     */
  },
  { virtual: true },
);
```

However, this rule falsely identifies these mocks as invalid.
